sources/lib/api/gcp/bigquery/_table.py

```diff
@@ -450,40 +459,14 @@ def load_async(self, source, mode='create', source_format='csv',
     Raises:
       Exception if the load job failed to be started or invalid arguments were supplied.
     """
-    if source_format == 'csv':
-      source_format = 'CSV'
-    elif source_format == 'json':
-      source_format = 'NEWLINE_DELIMITED_JSON'
-    else:
-      raise Exception("Invalid source format %s" % source_format)
-
-    if not(mode == 'create' or mode == 'append' or mode == 'overwrite'):
-      raise Exception("Invalid mode %s" % mode)
-
-    encoding_upper = encoding.upper()
-    if encoding_upper != 'UTF-8' and encoding_upper != 'ISO-8859-1':
-      raise Exception("Invalid source encoding %s" % encoding)
-
-    try:
-      response = self._api.jobs_insert_load(source, self._name_parts,
-                                            append=(mode == 'append'),
-                                            overwrite=(mode == 'overwrite'),
-                                            create=(mode == 'create'),
-                                            source_format=source_format,
-                                            field_delimiter=csv_delimiter,
-                                            allow_jagged_rows=allow_jagged_rows,
-                                            allow_quoted_newlines=allow_quoted_newlines,
-                                            encoding=encoding_upper,
-                                            ignore_unknown_values=ignore_unknown_values,
-                                            max_bad_records=max_bad_records,
-                                            quote=quote,
-                                            skip_leading_rows=csv_skip_header_rows)
-    except Exception as e:
-      raise e
-    return self._init_job_from_response(response)
+    csv_options = _csv_options.CSVOptions(csv_delimiter, csv_skip_leading_rows, encoding, quote,
+                                          allow_quoted_newlines, allow_jagged_rows)
+    dest = _external_table.ExternalTable(source, source_format, csv_options,
+                                         ignore_unknown_values, max_bad_records)
+    return dest.load_async(self, mode, self._context)
```

Comment:
To correspond with the comment there, I think this functionality belongs here, since its about actual BigQuery tables.